src/main/java/net/glowstone/net/codec/play/game/UpdateStructureBlockCodec.java

```diff
@@ -0,0 +1,88 @@
+package net.glowstone.net.codec.play.game;
+
+import com.flowpowered.network.Codec;
+import com.flowpowered.network.util.ByteBufUtils;
+import io.netty.buffer.ByteBuf;
+import net.glowstone.net.GlowBufUtils;
+import net.glowstone.net.message.play.game.UpdateStructureBlockMessage;
+import org.bukkit.util.BlockVector;
+
+import java.io.IOException;
+
+public final class UpdateStructureBlockCodec implements Codec<UpdateStructureBlockMessage> {
+    @Override
+    public UpdateStructureBlockMessage decode(ByteBuf byteBuf) throws IOException {
+        BlockVector location = GlowBufUtils.readBlockPosition(byteBuf);
+        int action = ByteBufUtils.readVarInt(byteBuf);
+        int mode = ByteBufUtils.readVarInt(byteBuf);
+        String name = ByteBufUtils.readUTF8(byteBuf);
+        byte offsetX = byteBuf.readByte();
+        byte offsetY = byteBuf.readByte();
+        byte offsetZ = byteBuf.readByte();
+        byte sizeX = byteBuf.readByte();
+        byte sizeY = byteBuf.readByte();
+        byte sizeZ = byteBuf.readByte();
+        int mirror = ByteBufUtils.readVarInt(byteBuf);
+        int rotation = ByteBufUtils.readVarInt(byteBuf);
+        String metadata = ByteBufUtils.readUTF8(byteBuf);
+        float integrity = byteBuf.readFloat();
+        long seed = ByteBufUtils.readVarLong(byteBuf);
+        byte flags = byteBuf.readByte();
+        boolean ignoreEntities = (flags & 1) == 1;
+        boolean showAir = ((flags >> 1) & 1) == 1;
+        boolean showBoundingBox = ((flags >> 3) & 1) == 1;
```

Comment:
You can simplify this with:
```suggestion
        boolean ignoreEntities = (flags & 1) == 1;
        boolean showAir = (flags & 2) == 1;
        boolean showBoundingBox = (flags & 4) == 1;
```

This makes it closer to the documentation of the bits (1, 2, 4).